src/commands/scaling/addScaleRule/AddNewScaleRule.ts

```diff
@@ -0,0 +1,86 @@
+import { localize } from "../../../utils/localize";
+import { updateContainerApp } from "../../updateContainerApp";
+import { IAddScaleRuleWizardContext } from "./IAddScaleRuleWizardContext";
+
+export class AddNewScaleRule extends AzureWizardExecuteStep<IAddScaleRuleWizardContext> {
+    public priority: number = 100;
+
+    public async execute(context: IAddScaleRuleWizardContext, _progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
+        const node: ScaleRuleGroupTreeItem = nonNullProp(context, "treeItem");
+        const containerApp: ContainerAppTreeItem = node.parent.parent instanceof RevisionTreeItem ? node.parent.parent.parent.parent : node.parent.parent;
+
+        const adding = localize('addingScaleRule', 'Adding scale rule setting to "{0}"...', containerApp.name);
+        const added = localize('addedScaleRule', 'Added scale rule setting to "{0}".', containerApp.name);
+
+        const template: Template = containerApp?.data?.template || {};
+        template.scale ||= {};
+        template.scale.rules ||= [];
+
+        const scaleRule: ScaleRule = this.buildScaleRule(context);
+        this.integrateRule(context, template.scale.rules, scaleRule);
+
+        await window.withProgress({ location: ProgressLocation.Notification, title: adding }, async (): Promise<void> => {
+            ext.outputChannel.appendLog(adding);
+            await updateContainerApp(context, containerApp, { template });
+
+            void window.showInformationMessage(added);
+            ext.outputChannel.appendLog(added);
+
+            await containerApp?.refresh(context);
+        });
+    }
+
+    public shouldExecute(): boolean {
+        return true;
+    }
+
+    private buildScaleRule(context: IAddScaleRuleWizardContext): ScaleRule {
+        const scaleRule: ScaleRule = { name: context.ruleName };
+        switch (context.ruleType) {
+            case ScaleRuleTypes.HTTP:
+                scaleRule.http = {
+                    metadata: {
+                        concurrentRequests: nonNullProp(context, 'concurrentRequests')
+                    }
+                };
+                break;
+            case ScaleRuleTypes.Queue:
+                scaleRule.azureQueue = {
+                    queueName: context.queueName,
+                    queueLength: Number(context.queueLength),
+                    auth: [{ secretRef: context.secretRef, triggerParameter: context.triggerParameter }]
+                }
+        }
+        return scaleRule;
+    }
+
+    private integrateRule(context: IAddScaleRuleWizardContext, scaleRules: ScaleRule[], scaleRule: ScaleRule): void {
+        switch (context.ruleType) {
+            case ScaleRuleTypes.HTTP:
+                const idx: number = scaleRules.findIndex((rule) => rule.http);
+                if (idx !== -1) {
+                    scaleRules[idx] = scaleRule;
+                } else {
+                    scaleRules.push(scaleRule);
```

Comment:
I think it makes sense to just have either a default case that always pushes the scaleRule in and instead of replacing the HTTP rule by overwriting `scaleRules[ind]`, to just splice it out.  That way you could do something more like

```
switch (context.ruleType) {
            case ScaleRuleTypes.HTTP:
                const idx: number = scaleRules.findIndex((rule) => rule.http);
                if (idx) {
                    scaleRules.splice(idx, 1)
                }
            case ScaleRuleTypes.Queue:
            default:
                   scaleRules.push(scaleRule);
}

```